### ISSUE TITLE
docker.tls.ca ignored when loading config from KV store

### ISSUE THREAD

**Issue by sjawhar** [containous/traefik#3249]
### Do you want to request a *feature* or report a *bug*?
Report a bug

### What did you do?
* Use `traefik storeconfig` to store config in consul
* Launch traefik with `--consul` to load config

### What did you expect to see?
`docker.tls.ca` setting loaded successfully

### What did you see instead?
`docker.tls.ca` is loaded as an empty string

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      v1.5.4
Codename:     cancoillotte
Go version:   go1.9.4
Built:        2018-03-15_01:35:21PM
OS/Arch:      linux/amd64
```

Also tried with 1.6.0-rc6

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
docker-compose.yml
```yaml
version: '3.5'
services:
  init:
    image: traefik:1.5.4
    command:
      - storeconfig
      - --defaultEntryPoints=http,https
      - --entrypoints=Name:http Address::80 Redirect.Entrypoint:https
      - --entrypoints=Name:https Address::443 Compress:true TLS
      - --entrypoints=Name:traefik Address::8080
      - --acme
      - --acme.entrypoint=https
      - --acme.storage=traefik/acme/account
      - --acme.email=admin@example.com
      - --acme.dnschallenge.provider=route53
      - --acme.onhostrule
      - --docker
      - --docker.swarmmode
      - --docker.watch
      - --docker.domain=example.com
      - --docker.exposedbydefault=false
      - --docker.endpoint=tcp://docker.socket:2376
      - --docker.tls
      - --docker.tls.ca=/run/secrets/socket-ca.pem
      - --docker.tls.cert=/run/secrets/socket-cert.pem
      - --docker.tls.key=/run/secrets/socket-key.pem
      - --ping
      - --consul
      - --consul.endpoint=consul.server:8500
      - --consul.prefix=traefik/v1
    deploy:
      restart_policy:
        condition: on-failure

  proxy:
    image: traefik:1.5.4
    env_file:
      - secrets.env
    networks:
      - default
      - reverse-proxy
    ports:
      - '80:80'
      - '443:443'
    secrets:
      - source: docker-socket-ca-cert-v1
        target: socket-ca.pem
      - source: docker-socket-traefik-cert-v1
        target: socket-cert.pem
      - source: docker-socket-traefik-key-v1
        target: socket-key.pem
    command:
      - --consul
      - --consul.endpoint=consul.server:8500
      - --consul.prefix=traefik/v1
```

### If applicable, please paste the log output at DEBUG level (`--logLevel=DEBUG` switch)
Here's the result of `traefik storeconfig`. Formatted for your eyeballs, and omitted irrelevant parts. Notice that CA has a value.
```
traefik_init.1.tzt8mwnwqy1c@linuxkit-00155d00681d    | 2018/04/29 09:53:00 Storing configuration:
{
  "LifeCycle":{
    "RequestAcceptGraceTimeout":0,
    "GraceTimeOut":10000000000
  },
  ...
  "Docker":{
    "Watch":true,
    "Filename":"",
    "Constraints":null,
    "Trace":false,
    "DebugLogGeneratedTemplate":false,
    "Endpoint":"tcp://docker.socket:2376",
    "Domain":"example.com",
    "TLS":{
      "CA":"/run/secrets/socket-ca.pem",
      "CAOptional":false,
      "Cert":"/run/secrets/socket-cert.pem",
      "Key":"/run/secrets/socket-key.pem",
      "InsecureSkipVerify":false
    },
    "ExposedByDefault":false,
    "UseBindPortIP":false,
    "SwarmMode":true
  },
  ...
  "Ping":{
    "EntryPoint":"traefik"
  }
}
```

Here's the output of the running traefik instance. Notice that CA is now an empty string.
```
traefik_proxy.1.sw95mqlige09@linuxkit-00155d00681d    | time="2018-04-29T10:19:22Z" level=info msg="Traefik version v1.6.0-rc6 built on 2018-04-17_11:59:31AM"
traefik_proxy.1.sw95mqlige09@linuxkit-00155d00681d    | time="2018-04-29T10:19:22Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
traefik_proxy.1.sw95mqlige09@linuxkit-00155d00681d    | time="2018-04-29T10:19:22Z" level=debug msg="Global configuration loaded 
{
  "LifeCycle":{
    "RequestAcceptGraceTimeout":0,
    "GraceTimeOut":10000000000
  },
  ...
  "Docker":{
    "Watch":true,
    "Filename":"",
    "Constraints":null,
    "Trace":false,
    "DebugLogGeneratedTemplate":false,
    "Endpoint":"tcp://docker.socket:2376",
    "Domain":"example.com",
    "TLS":{
      "CA":"",
      "CAOptional":false,
      "Cert":"/run/secrets/socket-cert.pem",
      "Key":"/run/secrets/socket-key.pem",
      "InsecureSkipVerify":false
    },
    "ExposedByDefault":false,
    "UseBindPortIP":false,
    "SwarmMode":true
  },
  ...
  "Ping":{
    "EntryPoint":"traefik"
  }
}
```

The value is being properly set by `storeconfig`:
![image](https://user-images.githubusercontent.com/5074378/39405775-be96b41a-4bab-11e8-8a73-bc1b8b05272b.png)

I'd be happy to address the bug myself, if someone could give me some pointers of where to look.

**Comment by sjawhar** [comment]
Manually changing ca to CA in the KV store fixes the problem. So there is some inconsistency between how `storeconfig` stores the values and how those values are then read back out.

**Comment by traefiker** [comment]
Closed by #3532.
